src/Microsoft.AspNetCore.OData.Authorization/ODataAuthorizationOptions.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.OData.Authorization
+{
+    /// <summary>
+    /// Provides configuration for the OData authorization layer
+    /// </summary>
+    public class ODataAuthorizationOptions
+    {
+        IServiceCollection _services;
+
+
+        public ODataAuthorizationOptions(IServiceCollection services)
+        {
+            _services = services;
```

Comment:
At the first look that seems unusal to me that the options class holds a reference to the ServiceCollection. 
I woudl think that has consequences on how these objects are allocated and deallocated.

HEre is an example, 
https://github.com/dotnet/aspnetcore/blob/0303c9e90b5b48b309a78c2ec9911db1812e6bf3/src/Mvc/Mvc/src/MvcServiceCollectionExtensions.cs#L44

using IServiceCollection.Configure<TOptions>(...) 
https://docs.microsoft.com/en-us/dotnet/api/microsoft.extensions.dependencyinjection.optionsconfigurationservicecollectionextensions.configure?view=dotnet-plat-ext-3.1#Microsoft_Extensions_DependencyInjection_OptionsConfigurationServiceCollectionExtensions_Configure__1_Microsoft_Extensions_DependencyInjection_IServiceCollection_System_String_Microsoft_Extensions_Configuration_IConfiguration_System_Action_Microsoft_Extensions_Configuration_BinderOptions__

Reply:
Hi @chrisspre I haven't yet figured out how to refactor this to avoid the `_services` field while still allowing the user to do:
```c#
services.AddOData().AddAuthorization(options => {
  options.ConfigureAuthentication(/*...*/); // this calls services.AddAuthentication() under the hood
});
```
Would it make a difference if the `options` object were short-lived?


It also appears that the MvcBuilder class in the aspnetcore code you referenced holds a references to the service collection in the builder.Services property. Or is this one a valid use case?

Reply:
@habbes Don't block on me. If MvcBuilder does the same, I am probably mistaken about the pattern how these options are to be used.